### PR TITLE
improve ASE traj

### PR DESCRIPTION
**Pull request by thangckt**
add functions to convert from others formats to ASE traj format

## Summary by CodeRabbit

- **New Features**
  - Enhanced system and labeled system handling with new parameters and functionalities.
  - Improved stress calculations by modifying method parameters.

- **Bug Fixes**
  - Corrected return types for several methods to ensure consistency and reliability.

- **Tests**
  - Added new test cases to validate system setups and trajectory file operations.

- **Chores**
  - Updated Python version matrix in workflow to include only version 3.11.

### CODE REVIEW

File: dpdata/plugins/ase.py [hunk 1]
```diff
@@ -112,7 +114,7 @@ def from_multi_systems(
         step: Optional[int] = None,
         ase_fmt: Optional[str] = None,
         **kwargs,
-    ) -> "ase.Atoms":
+    ) -> object:  # generator of "ase.Atoms"
```
Comment (njzjz) [inline dpdata/plugins/ase.py:117]: should use `typing.Generator["ase.Atoms"]`

File: dpdata/plugins/ase.py [hunk 2]
```diff
@@ -296,6 +298,24 @@ def from_labeled_system(
 
         return dict_frames
 
+    def to_system(self, data, **kwargs):
+        """Convert System to ASE Atoms object."""
+        list_atoms = ASEStructureFormat().to_system(data, **kwargs)
+        file_name = kwargs.get("file_name", "conf.traj")
```
Comment (njzjz) [inline dpdata/plugins/ase.py:304]: Could you explicitly write the argument in line 301 - the current code cannot generate docs

**Review by njzjz — CHANGES_REQUESTED** [review]

File: dpdata/plugins/ase.py (hunk 1 above)
Comment (thangckt) [reply dpdata/plugins/ase.py]: done

File: dpdata/plugins/ase.py (hunk 2 above)
Comment (thangckt) [reply dpdata/plugins/ase.py]: done

**Review by njzjz — APPROVED** [review]

File: dpdata/plugins/ase.py
Comment (njzjz) [inline dpdata/plugins/ase.py:1]: Could you please add UTs for the implementation? Thanks!

**Review by njzjz — CHANGES_REQUESTED** [review]

File: dpdata/plugins/ase.py [hunk 3]
```diff
@@ -140,7 +140,7 @@ def from_multi_systems(
         frames = ase.io.read(file_name, format=ase_fmt, index=slice(begin, end, step))
         yield from frames
```
Comment (link89) [inline dpdata/plugins/ase.py:141]: I guess you should use ase.io.iread or else yield from is meaningless. I will just load all data into memory.

File: dpdata/plugins/ase.py
Comment (thangckt) [reply dpdata/plugins/ase.py:1]: done

**Review by njzjz — APPROVED** [review]

**Pull request merged**